src/supercrafter333/BetterBan/EventListener.php

```diff
@@ -168,5 +168,12 @@ public function onLogin(PlayerLoginEvent $event) : void {
 			$player->kick($reason);
 			$event->cancel();
 		}
+		if (BetterBan::isBannedIp($player->getNetworkSession()->getIp())) {
+            $pl = BetterBan::getInstance();
+            $entry = $pl->useMySQL() ? $pl->getMySQLIpBans()->getEntry($player->getNetworkSession()->getIp()) : $pl->getServer()->getNameBans()->getEntry($player->getName());
+            $reason = str_replace(["{source}", "{expires}", "{reason}", "{line}"], [$entry->getSource(), $entry->getExpires() !== null ? BetterBan::getInstance()->toPrettyFormat($entry->getExpires(), BetterBan::getInstance()->getConfig("use-legacy-format", false)) : "Never", $entry->getReason(), "\n"], BetterBan::getInstance()->getConfig()->get("you-are-banned-ip-logout"));
+            $player->kick($reason);
```

Comment:
Please move this to `PlayerPreLoginEvent`, kick player by using `PlayerPreLoginEvent->setKickFlag()`, Just to improve the code.